modules/shell-session-vars.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  config = (
+    let
+      export = n: v: "export ${n}=\"${toString v}\"";
+      setenv = n: v: "setenv ${n} \"${toString v}\"";
+      shEnvVarsStr = concatStringsSep "\n" (
+        mapAttrsToList export config.home.sessionVariables
+      );
+      cshEnvVarsStr = concatStringsSep "\n" (
+        mapAttrsToList setenv config.home.sessionVariables
+      );
+    in mkIf (config.home.sessionVariableSetter == "shell"
+      || config.home.sessionVariableSetter == "bash") {
+      home.file.".local/share/home-manager/env.sh".text = ''
```

Comment:
Odd, I had a comment here about having
```
setVars = setter: concatStringsSep "\n" (
  mapAttrsToList setter config.home.sessionVariables
);
```
in the `let` section and then change this to
```
home.file.".local/share/home-manager/env.sh".text = setVars export;
home.file.".local/share/home-manager/env.csh".text = setVars setenv;
```
but I must have forgotten to press the "add comment" button. Doh!